src/lightning/pytorch/strategies/ddp.py

```diff
@@ -329,38 +329,6 @@ def reduce(
             return _sync_ddp_if_available(tensor, group, reduce_op=reduce_op)
         return tensor
 
-    def training_step(self, *args: Any, **kwargs: Any) -> STEP_OUTPUT:
```

Comment:
All step methods are now handled by the base strategy implementation and the individual _ForwardRedirection implementation. 

Basically, if the LM is wrapped, we redirect through forward, else we call the LM directly. This is equivalent to the previous logic for `if self.lightning_module.trainer.state.fn == TrainerFn.FITTING else ...` and so the code can be generalized like this for all strategies.